napari/layers/_shapes_layer/model.py

```diff
@@ -444,22 +499,30 @@ def block_update_properties(self):
     def _get_shape(self):
         """Determines the shape of the vertex data.
         """
-        if len(self.data._vertices) == 0:
-            return [1, 1]
+        if len(self._data_view._vertices) == 0:
+            slice_shape = [1, 1]
+        else:
+            slice_shape = list(np.max(self._data_view._vertices, axis=0) + 1)
+
+        if len(self.data.keys()) == 1:
+            return [1] * (self._input_ndim-2) + slice_shape
         else:
-            return np.max(self.data._vertices, axis=0) + 1
+            keys = list(self.data.keys())
+            keys.remove(())
+            max_val = np.array(keys).max(axis=0)
+            return list(max_val) + slice_shape
 
     @property
     def range(self):
         """list of 3-tuple of int: ranges of data for slicing specifed by
         (min, max, step).
         """
-        if len(self.data._vertices) == 0:
+        if len(self._data_view._vertices) == 0:
             maxs = [1, 1]
             mins = [0, 0]
         else:
-            maxs = np.max(self.data._vertices, axis=0) + 1
-            mins = np.min(self.data._vertices, axis=0)
+            maxs = np.max(self._data_view._vertices, axis=0) + 1
+            mins = np.min(self._data_view._vertices, axis=0)
 
         return [(min, max, 1) for min, max in zip(mins, maxs)]
```

Comment:
If this is only used for slicing, this could be `tuple(slice(min, max) for min, max in zip(mins, maxs))`.

Or if ~~you're~~ I'm really showing off, `tuple(itertools.starmap(slice, zip(mins, maxs)))`. =P

Reply:
I switched this to be a tuple, but the change you suggested caused errors in other places